src/hotspot/share/gc/shenandoah/shenandoahHeap.hpp

```diff
@@ -524,6 +527,10 @@ class ShenandoahHeap : public CollectedHeap {
   // Prepare and finish concurrent unloading
   void prepare_concurrent_unloading();
   void finish_concurrent_unloading();
+  // Heap iteration support
+  void scan_roots_for_iteration(Stack<oop, mtGC>* oop_stack, ObjectIterateScanRootClosure* oops);
```

Comment:
Let's `typedef Stack<oop, mtGC> ShenandoahScanRootStack` and use it everywhere?

Reply:
As this Stack is also used for iterating all objects (not only roots), so I took the liberty to rename it to ShenandoahScanObjectStack :)